### PR TITLE
gb-vendor: Order dependencies in manifest file consistently

### PR DESCRIPTION
**Pull request by euank**
Prior to this change, any time you ran `gb vendor update $dependency` the dependency in question would be moved to the bottom of the manifest, regardless of any other changes. This behavior also happened when adding and removing dependencies.
This led to a noisy diff where it was difficult to easily see whether a commit hash changed or if the dependency moved etc.
This change alphabetically sorts dependencies within the manifest file, thus making the git diff of revision changes, additions, and removals much cleaner looking and easier to understand.

### CODE REVIEW

**Comment by davecheney** [comment]
LGTM with one change.

File: cmd/gb-vendor/vendor/manifest.go [hunk 1]
```diff
@@ -137,3 +141,14 @@ func readManifest(r io.Reader) (*Manifest, error) {
 	err := d.Decode(&m)
 	return &m, err
 }
+
+// Implement sort for dependencies
+type sortableDependencies []Dependency
```
Comment (davecheney) [inline cmd/gb-vendor/vendor/manifest.go]: please call this byImportpath (or ByImportpath) so you get 

     sort.Sort(byImportpath(m.Dependencies))

**Comment by davecheney** [comment]
Thanks. If you can address this one little nit then i'll merge this asap

**Comment by euank** [comment]
Good suggestion! Done and force-pushed over the old commit.

Thanks!

File: cmd/gb-vendor/vendor/manifest.go [hunk 2]
```diff
@@ -137,3 +141,14 @@ func readManifest(r io.Reader) (*Manifest, error) {
 	err := d.Decode(&m)
 	return &m, err
 }
+
+// Implement sort for dependencies
```
Comment (davecheney) [inline cmd/gb-vendor/vendor/manifest.go]: // byImport implements sort for dependencies.

**Comment by davecheney** [comment]
LGTM. Thanks for your contribution.

File: cmd/gb-vendor/vendor/manifest.go (hunk 2 above)
Comment (euank) [inline cmd/gb-vendor/vendor/manifest.go]: I'm a little lazy for unexported types. Feel free to fix it or I could go back and do so if you feel strongly.

**Pull request merged**